packages/SwingSet/src/kernel/vatManager/supervisor-helper.js

```diff
@@ -191,3 +191,45 @@ function makeSupervisorSyscall(syscallToManager, workerCanBlock) {
 
 harden(makeSupervisorSyscall);
 export { makeSupervisorSyscall };
+
+/**
+ * Create a vat console, or a no-op if consensusMode is true.
+ *
+ * TODO: consider other methods per SES VirtualConsole.
+ * See https://github.com/Agoric/agoric-sdk/issues/2146
+ *
+ * @param {Record<'debug' | 'log' | 'info' | 'warn' | 'error', (...args: any[]) => void>} logger
+ * the backing log method
+ * @param {boolean | ((logger: any, args: any[]) => void)} [wrapper]
+ */
+function makeVatConsole(logger, wrapper = true) {
+  const cons = Object.fromEntries(
+    ['debug', 'log', 'info', 'warn', 'error'].map(level => {
+      if (wrapper === false) {
+        // Static wrapper that never enables.  We create unique no-op functions
+        // so that the different log methods cannot be compared to detect this
+        // mode.
+        return [level, () => {}];
+      }
+
+      const backingLog = logger[level];
+      if (wrapper === true) {
+        // Static wrapper that always enables.
+        return [level, backingLog];
+      }
+
+      // Dynamic wrapper that may change its mind.
+      assert.typeof(
+        wrapper,
+        'function',
+        X`Invalid VatConsole wrapper value ${wrapper}`,
+      );
+      return [level, (...args) => wrapper(backingLog, args)];
```

Comment:
Something feels funny here, why have a three-way switch here (always-enabled, never-enabled, switchable) when `supervisor-subprocess-xsnap.js` also has a switch? This feels like maybe we could omit the switching code here and have the supervisors manage the disabling instead.

Reply:
The identity of the `console` is permanently fixed in the case of snapshots, so we need this switching code to be in the same place as where the `console` is first defined.  I could just always enable the switching code and not use the always/never modes (which you maybe consider to be premature optimisation).

Reply:
yeah, I think I prefer the simplicity of having one mode, over the nominal performance improvement of sometimes omitting the switch